src/main/java/moze_intel/projecte/network/commands/EMCCMD.java

```diff
@@ -0,0 +1,157 @@
+import moze_intel.projecte.api.capabilities.PECapabilities;
+import moze_intel.projecte.utils.TransmutationEMCFormatter;
+import moze_intel.projecte.utils.text.PELang;
+import net.minecraft.ChatFormatting;
+import net.minecraft.commands.CommandBuildContext;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.arguments.EntityArgument;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.server.level.ServerPlayer;
+
+import javax.annotation.Nullable;
+import java.math.BigInteger;
+
+public class EMCCMD {
+    private enum ActionType {
+        ADD,
+        REMOVE,
+        SET,
+        GET,
+        TEST
+    }
+    public static ArgumentBuilder<CommandSourceStack, ?> register(CommandBuildContext context) {
+         return Commands.literal("emc")
+            .requires(PEPermissions.COMMAND)
+            .then(Commands.literal("add")
+                .requires(PEPermissions.COMMAND_EMC_ADD)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.ADD))
+                    )
+                )
+            )
+            .then(Commands.literal("remove")
+                .requires(PEPermissions.COMMAND_EMC_REMOVE)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.REMOVE))
+                    )
+                )
+            )
+            .then(Commands.literal("set")
+                .requires(PEPermissions.COMMAND_EMC_SET)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.SET))
+                    )
+                )
+            )
+            .then(Commands.literal("test")
+                .requires(PEPermissions.COMMAND_EMC_TEST)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .then(Commands.argument("value", StringArgumentType.string())
+                        .executes((ctx) -> handle(ctx, ActionType.TEST))
+                    )
+                )
+            )
+            .then(Commands.literal("get")
+                .requires(PEPermissions.COMMAND_EMC_GET)
+                .then(Commands.argument("player", EntityArgument.player())
+                    .executes((ctx) -> handle(ctx, ActionType.GET))
+                )
+            );
+    }
+
+    private static MutableComponent formatEMC(BigInteger emc) {
+        return MutableComponent.create(TransmutationEMCFormatter.formatEMC(emc).getContents()).withStyle(ChatFormatting.GRAY);
```

Comment:
I believe `TextComponentUtil.build(ChatFormatting.GRAY, TransmutationEMCFormatter.formatEMC(emc))` should work, alternatively this could in theory use `TransmutationEMCFormatter.formatEMC(emc).copy().withStyle(ChatFormatting.GRAY)`